juju/juju
# feat: inject app service for charm retrieval into legacy state

This patch wires the domain services for charm retrieval into the legacy state. It is an important milestone because it allows us to start cleaning up the legacy state regarding charms.

Many small changes were needed to adapt the output of the legacy charm retrieval with the new domain and all its call sites, among them:
- `Version()` was added to the interfaces **and** as a new field in the CharmInfo returned RPC params.
- `IsAvailable` was added as a new return value on `GetCharm()`. This was needed for the object store to return the correct error when the charm was already available.

Also, `UpdateUploadedCharm()` was removed since no longer needed now that we are reading from dqlite.

## QA steps

Bootstrap and deploy ubuntu should continue working. We are aware of local charms not working, and more complex scenarios will be tested and fixed when we continue removing state.

```
$ juju bootstrap lxd c
$ juju add-model m
$ juju deploy ubuntu
```

## Links

**Jira card:** JUJU-6846
